runtime/platform/targets.bzl

```diff
@@ -41,10 +41,12 @@ def define_common_targets():
     # client defined implementations will overide them.
     runtime.cxx_library(
         name = "platform_private",
-        srcs = _select_pal({
-            "minimal": ["default/minimal.cpp"],
-            "posix": ["default/posix.cpp"],
-        }),
+        srcs = select({
+            "ovr_config//os:android": ["default/android.cpp"],
+            "DEFAULT": _select_pal({
+                "minimal": ["default/minimal.cpp"],
+                "posix": ["default/posix.cpp"],
```

Comment:
I'm wondering if this should be another "android" entry

Reply:
> However, it might also cause some confusion for users since executorch.pal_default is now ignored

That makes sense, but ovr_config//os:android is automatic.